astroid/nodes/_base_nodes.py

```diff
@@ -220,3 +247,524 @@ def _elsed_block_range(
+                )
+                if not_implemented and not_implemented != len(results):
+                    # Can't infer yet what this is.
+                    yield util.Uninferable
+                    return
+
+                yield from results
+                return
+
+        yield util.Uninferable
+
+
+class AttributeNode(NodeNG):
+    expr: NodeNG
+    """The name that this node represents."""
+
+    @decorators.raise_if_nothing_inferred
+    def _infer_attribute(
+        self,
+        context: InferenceContext | None = None,
+        **kwargs: Any,
+    ) -> Generator[InferenceResult, None, InferenceErrorInfo]:
+        """Infer an Attribute node by using getattr on the associated object."""
+        # pylint: disable=import-outside-toplevel
+        from astroid.constraint import get_constraints
+
+        for owner in self.expr.infer(context):
+            if isinstance(owner, util.UninferableBase):
+                yield owner
+                continue
+
+            context = copy_context(context)
+            old_boundnode = context.boundnode
+            try:
+                context.boundnode = owner
+                if isinstance(owner, (nodes.ClassDef, bases.Instance)):
+                    frame = (
+                        owner if isinstance(owner, nodes.ClassDef) else owner._proxied
+                    )
+                    context.constraints[self.attrname] = get_constraints(
+                        self, frame=frame
+                    )
+                yield from owner.igetattr(self.attrname, context)
+            except (
+                AttributeInferenceError,
+                InferenceError,
+                AttributeError,
+            ):
+                pass
+            finally:
+                context.boundnode = old_boundnode
+        return InferenceErrorInfo(node=self, context=context)
+
+
+class AssignNode(NodeNG):
+    @decorators.raise_if_nothing_inferred
+    @decorators.path_wrapper
+    def _infer_assign(
+        self,
+        context: InferenceContext | None = None,
+        **kwargs: Any,
+    ) -> Generator[InferenceResult, None, None]:
+        """Infer a AssignName/AssignAttr: need to inspect the RHS part of the
+        assign node.
+        """
+        if isinstance(self.parent, nodes.AugAssign):
+            return self.parent.infer(context)
+
+        stmts = list(self.assigned_stmts(context=context))
+        return bases._infer_stmts(stmts, context)
+
+    def assigned_stmts(
+        self: nodes.AssignName | nodes.AssignAttr,
+        node: AssignedStmtsPossibleNode = None,
+        context: InferenceContext | None = None,
+        assign_path: list[int] | None = None,
+    ) -> Any:
+        """Returns the assigned statement (non inferred) according to the assignment type."""
+        return self.parent.assigned_stmts(node=self, context=context)
+
+
+class NameNode(LookupMixIn):
+    name: str
```

Comment:
See comment above.

Reply:
The concrete classes for this one lack a postinit, though!

Reply:
fb05e0c65e443e0e438142821ba120680ac0dbff

Reply:
Do they need one? `name` gets set in the `init`?

Reply:
Whoops, you're totally right. Reverting.